README.md
<img src="https://www.forumcinemas.lt/files/fclogo/naujas/FC_logo_3.png" alt="ForumCinemas logo" width="200" height="100">

# Forum Cinemas Webscraper
A simple Python 3 webscraper for forumcinemas.lt. Gets upcoming movie data (name, director, duration, genre, image link, etc.) and data in the news section (title, description and link of the news). The data is saved to data.db file, as well as JSON files in package/data.

Uses requests and BeautifulSoup packages. Features regular, as well as multithreaded/multiprocessed versions of webscraping (for better performance).

# Instructions
Install requests and BeautifulSoup packages via pip.
```
pip install requests bs4
```
Then, simply run the webscraper.py file.
```
python3 webscraper.py
```

You can run either the regular, multithreaded or multiprocessed versions (or all of them, to compare performance and operation times) by typing in "regular", "multithreaded", "multiprocessed" or "all" into the console respectively (without quotation marks).


CHANGELOG.md
# Changelog

## [1.0.0] - 2022-01-18
  
### Added
- Added webscraper.
### Changed

### Fixed


webscraper.py
from package import movie
from package import news
import requests
from datetime import datetime
from bs4 import BeautifulSoup
from threading import Thread, Lock
from multiprocessing import Process
import filereadwrite
import time

movie_count = 0
news_count = 0
regular_time = 0
multithreaded_time = 0
multiprocessed_time = 0
site_movies_coming_soon = 'http://www.forumcinemas.lt/Movies/ComingSoon/'
site_news = 'http://www.forumcinemas.lt/news/kinonaujienos/'
news_result = requests.get(site_news)
movie_result = requests.get(site_movies_coming_soon)
news_r = news_result.content
movie_r = movie_result.content
movie_soup = BeautifulSoup(movie_r, "html.parser")
news_soup = BeautifulSoup(news_r, "html.parser")
movie_samples = movie_soup.find_all("a", "arrowLink")
news_samples = news_soup.find_all("a", "arrowLink")
data = {}
data_2 = {}


def scrape_upcoming_movies():
    global movie_count
    for a in movie_samples:
        if not a.string is None:
            if 'href' in a.attrs.keys():
                data[a] = a.attrs['href']
    for item in data:
        title_lt = "None"
        title_foreign = "None"
        age = "None"
        duration = "None"
        director_name = "None"
        image = "None"
        genre = "None"
        is_3d = False
        site_result = requests.get(data[item])
        re = site_result.content
        soup_coming_soon = BeautifulSoup(re, "html.parser")
        information_samples = soup_coming_soon.find_all("div", attrs={'id': "eventInfoBlock"})
        for a in information_samples:
            image = a.img['src']
            text = a.text.strip()
            text = text.split("\n")
            text = [w.replace('\r', '') for w in text]
            for item in text:
                text[text.index(item)] = text[text.index(item)].lstrip()
            text = list(filter(None, text))
            title_lt = text[0]
            title_foreign = text[1]
            title_foreign = title_foreign.lstrip()
            duration = text[text.index("Trukmė: ") + 1] + " " + text[text.index("Trukmė: ") + 2] + " " + text[
                text.index("Trukmė: ") + 3]
            if "Žanras:" in text and "Sukurta:" in text:
                genre = (text[(text.index("Žanras:") + 1):text.index("Sukurta:")])
                genre = ''.join(genre)
            if "Režisierius:" in text and "Vaidina:" in text:
                director_name = (text[(text.index("Režisierius:") + 1):text.index("Vaidina:")])
                director_name = ''.join(director_name)
            if "Režisierius:" in text and "Vaidina:" not in text:
                director_name = text[(text.index("Režisierius:") + 1)]
            if "3D" in title_lt or "3D" in title_foreign:
                is_3d = True
            age = text[text.index("Cenzas: ") + 1]
            if age == "Įvairaus amžiaus žiūrovams":
                age = 0
            if age == "N-7. Jaunesniems būtina suaugusiojo palyda":
                age = 7
            if age == "N-13. 7-12 m. vaikams būtina suaugusiojo palyda":
                age = 13
            if age == "Žiūrovams nuo 16 metų":
                age = 16
            if age == "Suaugusiems nuo 18 metų":
                age = 18
        lock = Lock()
        lock.acquire()
        movie.add_movie(title_lt, title_foreign, director_name, age, duration, is_3d, image, genre)
        lock.release()
        movie_count = movie_count + 1


# /----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
def scrape_news_and_additional_movies():
    global movie_count, news_count
    for b in news_samples:
        if not b.string is None:
            if 'href' in b.attrs.keys():
                data_2[b] = b.attrs['href']

    for item in data_2:
        if ".lt/event" in data_2[item]:
            title_lt = "None"
            title_foreign = "None"
            age = "None"
            duration = "None"
            director_name = "None"
            image = "None"
            genre = "None"
            is_3d = False
            site_result = requests.get(data_2[item])
            re = site_result.content
            soup_coming_soon = BeautifulSoup(re, "html.parser")
            information_samples = soup_coming_soon.find_all("div", attrs={'id': "eventInfoBlock"})
            for a in information_samples:
                image = a.img['src']
                text = a.text.strip()
                text = text.split("\n")
                text = [w.replace('\r', '') for w in text]
                for item in text:
                    text[text.index(item)] = text[text.index(item)].lstrip()
                text = list(filter(None, text))
                title_lt = text[0]
                title_foreign = text[1]
                title_foreign = title_foreign.lstrip()
                duration = text[text.index("Trukmė: ") + 1] + " " + text[text.index("Trukmė: ") + 2] + " " + text[
                    text.index("Trukmė: ") + 3]
                if "Žanras:" in text and "Sukurta:" in text:
                    genre = (text[(text.index("Žanras:") + 1):text.index("Sukurta:")])
                    genre = ''.join(genre)
                if "Režisierius:" in text and "Vaidina:" in text:
                    director_name = (text[(text.index("Režisierius:") + 1):text.index("Vaidina:")])
                    director_name = ''.join(director_name)
                if "Režisierius:" in text and "Vaidina:" not in text:
                    director_name = text[(text.index("Režisierius:") + 1)]
                if "3D" in title_lt or "3D" in title_foreign:
                    is_3d = True
                age = text[text.index("Cenzas: ") + 1]
                if age == "Įvairaus amžiaus žiūrovams":
                    age = 0
                if age == "N-7. Jaunesniems būtina suaugusiojo palyda":
                    age = 7
                if age == "N-13. 7-12 m. vaikams būtina suaugusiojo palyda":
                    age = 13
                if age == "Žiūrovams nuo 16 metų":
                    age = 16
                if age == "Suaugusiems nuo 18 metų":
                    age = 18
            lock = Lock()
            lock.acquire()
            movie.add_movie(title_lt, title_foreign, director_name, age, duration, is_3d, image, genre)
            lock.release()
            movie_count = movie_count + 1
        elif ".lt/news" in data_2[item]:
            link = data_2[item]
            site_result = requests.get(data_2[item])
            re = site_result.content
            soup_coming_soon = BeautifulSoup(re, "html.parser")
            information_samples = soup_coming_soon.find_all("div", attrs={'class': "contboxBasic"})
            text = information_samples[0].text.strip()
            text = text.split("\n")
            text = [w.replace('\r', '') for w in text]
            for item in text:
                text[text.index(item)] = text[text.index(item)].lstrip()
            text = list(filter(None, text))
            title = text[1]
            try:
                description = text[3]
            except IndexError:
                description = "None"
            lock = Lock()
            lock.acquire()
            news.add_news(title, description, link)
            lock.release()
            news_count = news_count + 1

    print("Total movie count: ", movie_count)
    print("Total news count: ", news_count)


# ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
def webscraping_regular():
    global regular_time
    global movie_count
    global news_count
    print("Starting regular web scraper in 5 seconds...")
    time.sleep(5)
    start_time = datetime.now()
    print(f'[{start_time}] {"Starting!"}')
    scrape_upcoming_movies()
    scrape_news_and_additional_movies()
    end_time = datetime.now()
    print(f'[{datetime.now()}] {"Regular web scraping ending!"}')
    regular_time = end_time - start_time
    movie_count = 0
    news_count = 0


def webscraping_multithreaded():
    global multithreaded_time
    global movie_count
    global news_count
    print("Starting multithreaded web scraper in 5 seconds...")
    time.sleep(5)
    start_time = datetime.now()
    print(f'[{start_time}] {"Starting!"}')
    Thread(target=scrape_upcoming_movies()).start()
    Thread(target=scrape_news_and_additional_movies()).start()
    end_time = datetime.now()
    print(f'[{end_time}] {"Multithreading web scraping ending!"}')
    multithreaded_time = end_time - start_time
    movie_count = 0
    news_count = 0


def webscraping_multiprocessed():
    global multiprocessed_time
    print("Starting multiprocessed web scraper in 5 seconds...")
    time.sleep(5)
    start_time = datetime.now()
    print(f'[{start_time}] {"Starting!"}')
    Process(target=scrape_upcoming_movies()).start()
    Process(target=scrape_news_and_additional_movies()).start()
    end_time = datetime.now()
    print(f'[{end_time}] {"Multiprocessing web scraping ending!"}')
    multiprocessed_time = end_time - start_time


print(
    "Welcome! Would you like to start the regular, multithreaded or multiprocessed version of the web scraper, or all of them?")
mode = input('Please input (without quotation marks) "regular", "multithreaded", "multiprocessed" or "all":   ')
if mode == "regular":
    webscraping_regular()
    print("Regular scraping completed in: ", regular_time)
if mode == "multithreaded":
    webscraping_multithreaded()
    print("Multithreaded scraping completed in: ", multithreaded_time)
if mode == "multiprocessed":
    webscraping_multiprocessed()
    print("Multiprocessed scraping completed in: ", multiprocessed_time)
if mode == "all":
    webscraping_regular()
    filereadwrite.File.file_delete("movies")
    filereadwrite.File.file_delete("news")
    webscraping_multithreaded()
    filereadwrite.File.file_delete("movies")
    filereadwrite.File.file_delete("news")
    webscraping_multiprocessed()
    print("Regular scraping completed in: ", regular_time)
    print("Multithreaded scraping completed in: ", multithreaded_time)
    print("Multiprocessed scraping completed in: ", multiprocessed_time)


package/news.py
import filereadwrite
import sqlite3

# News may have a title, a description and a link.

class News:

    def __init__(self, title, description, link):
        self.title = title
        self.description = description
        self.link = link

    # Get the title of the news.
    def news_name(self):
        return self.title

    # Get the description of the news.
    def news_description(self):
        return self.description

    # Get the link of the news.
    def news_link(self):
        return self.link

    # Get all the attributes of the news.

    def __str__(self):
        return "News title: ", self.title, ", news description: ", self.description, ", news link: ", self.link


# News, along with their title and description are added.
def add_news(added_title, added_description, added_link):
    conn = sqlite3.connect('data.db')
    c = conn.cursor()
    c.execute('''CREATE TABLE IF NOT EXISTS news (id INTEGER PRIMARY KEY AUTOINCREMENT, title VARCHAR(1000), description VARCHAR(10000),
        link VARCHAR(300))''')
    newsfile = filereadwrite.File.file_read_create_if_not_found("news")
    news_dict = dict()
    for item in newsfile:
        n = News(newsfile[item]["title"], newsfile[item]["description"], newsfile[item]["link"])
        news_dict[item] = vars(n)
    n = News(added_title, added_description, added_link)
    news_dict[added_title] = vars(n)
    filereadwrite.File.file_write("news", news_dict)
    print("News found and added! Title:", n.news_name(), ", description: ", n.news_description(), ", link: ",
          n.news_link())
    c.execute('SELECT * FROM news WHERE (title=? AND description=? AND link=?)', (n.news_name(), n.news_description(),n.news_link()))
    entry = c.fetchone()
    if entry is None:
        conn.execute(
            'INSERT INTO news (title, description, link) VALUES (?,?,?)',
            (n.news_name(), n.news_description(), n.news_link()))
    else:
        pass
    conn.commit()
    conn.close()


#  News selected by the title are deleted.
def remove_news(removed_title):
    newsfile = filereadwrite.File.file_read("news")
    news_dict = dict()
    news_deleted = False
    for item in newsfile:
        n = News(newsfile[item]["title"], newsfile[item]["description"], newsfile[item]["link"])
        if removed_title == n.news_name():
            news_deleted = True
            print("News found and removed! Title:", n.news_name(), ", description: ", n.news_description(), ", link: ",
                  n.news_link())
            pass
        else:
            news_dict[item] = vars(n)
    filereadwrite.File.file_write("news", news_dict)
    if news_deleted == False:
        print("No news with such name found, therefore no news have been deleted.")


# All the news are printed.
def output_news():
    newsfile = filereadwrite.File.file_read("news")
    news_dict = dict()
    for item in newsfile:
        n = News(newsfile[item]["title"], newsfile[item]["description"], newsfile[item]["link"])
        news_dict[item] = vars(n)
        print(vars(n), "\n")
    filereadwrite.File.file_write("results", news_dict)


filereadwrite.py
import os
import json


class File:
    def file_read(filename):
        if os.stat('package/data/' + filename + '.json').st_size == 0:
            print(filename + ".json file is empty!")
            exit(0)
        elif os.stat('package/data/' + filename + '.json').st_size != 0:
            with open('package/data/' + filename + '.json', encoding=('UTF-8')) as f:
                file = json.load(f)
        return file


    def file_read_create_if_not_found(filename):
        if not os.path.isfile('package/data/' + filename + '.json'):
            print(filename, ".json file not found! Creating one now...")
            File.file_write(filename, {})
            with open('package/data/' + filename + '.json', encoding=('UTF-8')) as f:
                file = json.load(f)
        else:
            with open('package/data/' + filename + '.json', encoding=('UTF-8')) as f:
                file = json.load(f)
        return file

    def file_write(filename, item):  # Overwrites file!
        with open('package/data/' + filename + '.json', 'w', encoding=('UTF-8')) as outfile:
            json.dump(item, outfile, ensure_ascii=False)

    def file_delete(filename):
        os.remove('package/data/' + filename + '.json')


package/ticket.py
import filereadwrite
import datetime
from package import movie


# Ticket may have a movie title, price, seat number, row number, date, movie beginning time and movie ending time.

class Ticket:
    def __init__(self, title, price, seat_number, row_number, date, movie_start_time, movie_end_time):
        self.title = title
        self.price = price
        self.seat_number = seat_number
        self.row_number = row_number
        self.date = date
        self.movie_start_time = movie_start_time
        self.movie_end_time = movie_end_time

    # Get the title of the movie which the ticket is bought for.

    def ticket_name(self):
        return self.title

    # Get the price of the ticket.

    def ticket_price(self):
        return self.price

    # Get the seat number of the ticket.

    def ticket_seat_number(self):
        return self.seat_number

    # Get the date of the ticket.

    def ticket_date(self):
        return self.date

    # Get the movie beginning time.

    def ticket_movie_start_time(self):
        return self.movie_start_time

    # Get the movie ending time.

    def ticket_movie_end_time(self):
        return self.movie_end_time

    # Get all the attributes of the ticket.

    def __str__(self):
        return "Ticket movie name: ", self.title, ", ticket price: ", self.price, ", seat number: ", self.seat_number, ", ticket date: ", self.date, ", movie start time: ", self.movie_start_time, ", movie end time: ", self.movie_end_time


# Purchase a ticket.
def buy_ticket(moviename, time, seat, row):
    moviefile = filereadwrite.File.file_read("movies")
    movie_found = False
    ticket_dict = dict()
    ticketfile = filereadwrite.File.file_read_create_if_not_found("tickets")
    for item in ticketfile:
        t = Ticket(ticketfile[item]["title"], ticketfile[item]["price"], ticketfile[item]["seat_number"],
                   ticketfile[item]["row_number"], ticketfile[item]["date"], ticketfile[item]["movie_start_time"],
                   ticketfile[item]["movie_end_time"])
        ticket_dict[item] = vars(t)
    for item in moviefile:
        m = movie.Movie(moviefile[item]["title"], moviefile[item]["foreign_title"], moviefile[item]["director"],
                        moviefile[item]["age_rating"],
                        moviefile[item]["duration"], moviefile[item]["is_3d"])
        timestring = moviefile[item]["duration"]
        hours = int(timestring[0:(timestring.index("h"))])
        minutes = int(timestring[timestring.index("h") + 1: timestring.index("min")])
        movie_end_time = datetime.datetime.strptime(time, '%H:%M') + datetime.timedelta(
            minutes=((hours * 60) + minutes))
        movie_end_time = str(movie_end_time.time())
        if m.movie_is_3d():
            price = 5
        else:
            price = 4.5
        if m.movie_name() == moviename or m.movie_foreign_title() == moviename:
            movie_found = True
            t = Ticket(m.movie_name(), price, seat, row, str(datetime.date.today()), time, movie_end_time)
            ticket_dict[item] = vars(t)
            print("Ticket bought! Movie title: ", t.ticket_name(), ", ticket price: ", t.ticket_price(),
                  ", seat number: ", t.ticket_seat_number(), ", movie start time: ", t.ticket_movie_start_time(),
                  ", movie end time: ", t.ticket_movie_end_time())
        filereadwrite.File.file_write("tickets", ticket_dict)
    if not movie_found:
        print("No movies with such name found!")


package/movie.py
import filereadwrite
from package import news
import sqlite3

class Movie:

    def __init__(self, title, foreign_title, director, age_rating, duration, is_3d, image, genre):
        self.title = title
        self.foreign_title = foreign_title
        self.director = director
        self.age_rating = age_rating
        self.duration = duration
        self.is_3d = is_3d
        self.image = image
        self.genre = genre

    # Get the title of the movie.
    def movie_name(self):
        return self.title

    # Get the 3D status of the movie.

    def movie_foreign_title(self):
        return self.foreign_title

    # Get the director of the movie.

    def movie_director(self):
        return self.director

    # Get the age-rating of the movie.
    def movie_age_rating(self):
        return self.age_rating

    # Get the duration of the movie.
    def movie_duration(self):
        return self.duration

    # Get the 3D status of the movie.

    def movie_is_3d(self):
        return self.is_3d

    # Get the image of the movie.

    def movie_image(self):
        return self.image
		
	# Get the genre of the movie.
    def movie_genre(self):
        return self.genre
		
    # Get all the attributes of the movie.

    def __str__(self):
        return "Movie title: ", self.title, " director: ", self.director, ", age rating: ", self.age_rating, "duration in minutes: ", self.duration, ", the movie is in 3D: ", self.is_3d, ", image link: ", self.image, ", genre: ", self.genre


# A new movie is added to the movies.json file.
def add_movie(added_title, added_foreign_title, added_director, added_age_rating, added_duration, added_is_3d, added_image, added_genre):
    conn = sqlite3.connect('data.db')
    c = conn.cursor()
    c.execute('''CREATE TABLE IF NOT EXISTS movies (id INTEGER PRIMARY KEY AUTOINCREMENT, title VARCHAR(100), foreign_title VARCHAR(100),
    director VARCHAR(300), age_rating INTEGER, duration VARCHAR(10), is_3d BOOLEAN, image VARCHAR(200), genre VARCHAR(300))''')
    movie_file = filereadwrite.File.file_read_create_if_not_found("movies")
    movie_dict = dict()
    for item in movie_file:
        m = Movie(movie_file[item]["title"], movie_file[item]["foreign_title"], movie_file[item]["director"],
                  movie_file[item]["age_rating"],
                  movie_file[item]["duration"], movie_file[item]["is_3d"], movie_file[item]["image"], movie_file[item]["genre"])
        if m.movie_name() == added_title or m.movie_foreign_title() == added_foreign_title:
            print("Movie already exists!")
            pass
        movie_dict[item] = vars(m)
    if added_is_3d == "true" or added_is_3d == "True" or added_is_3d == "yes" or added_is_3d == "Yes":
        added_is_3d = True
    if added_is_3d == "false" or added_is_3d == "False" or added_is_3d == "no" or added_is_3d == "No":
        added_is_3d = False
    m = Movie(added_title, added_foreign_title, added_director, added_age_rating, added_duration, added_is_3d, added_image, added_genre)
    movie_dict[m.movie_name()] = vars(m)
    filereadwrite.File.file_write("movies", movie_dict)
    print("Movie added! Title:", m.movie_name(), ", foreign title: ", m.movie_foreign_title(), ", director: ",
          m.movie_director(), ", age rating: ", m.movie_age_rating(), ", duration: ", m.movie_duration(),
          ", is in 3D: ", m.movie_is_3d(), ", image link: ", m.movie_image(), ", genre: ", m.movie_genre())
    c.execute('SELECT * FROM movies WHERE (title=? AND foreign_title=? AND director=? AND age_rating=? AND duration=? AND is_3d=? AND genre=?)',
              (m.movie_name(), m.movie_foreign_title(), m.movie_director(), m.movie_age_rating(), m.movie_duration(),
                  m.movie_is_3d(), m.movie_genre()))
    entry = c.fetchone()
    if entry is None:
        conn.execute(
            'INSERT INTO movies (title, foreign_title, director, age_rating, duration, is_3d, image, genre) VALUES (?,?,?,?,?,?,?,?)', (
            m.movie_name(), m.movie_foreign_title(), m.movie_director(), m.movie_age_rating(), m.movie_duration(),
            m.movie_is_3d(), m.movie_image(), m.movie_genre()))
    else:
        pass
    conn.commit()
    conn.close()


# A specified movie is removed from the movies.json file.
def remove_movie(removed_title):
    movie_file = filereadwrite.File.file_read("movies")
    movie_dict = dict()
    movie_deleted = False
    for item in movie_file:
        m = Movie(movie_file[item]["title"], movie_file[item]["foreign_title"], movie_file[item]["director"],
                  movie_file[item]["age_rating"],
                  movie_file[item]["duration"], movie_file[item]["is_3d"], movie_file[item]["image"])
        if removed_title == m.movie_name() or removed_title == m.movie_foreign_title():
            movie_deleted = True
            pass
        else:
            movie_dict[item] = vars(m)
    filereadwrite.File.file_write("movies", movie_dict)
    if not movie_deleted:
        print("No movie with such name found, therefore no movies have been deleted.")
    else:
        print("Movie removed! Title:", m.movie_name(), ", foreign title: ", m.movie_foreign_title(), ", director: ",
              m.movie_director(), ", age rating: ", m.movie_age_rating(), ", duration: ", m.movie_duration(),
              ", is in 3D: ", m.movie_is_3d(), ", image link: ", m.movie_image())


# Only movies that are in 3D are printed.
def output_3d_movies():
    movie_file = filereadwrite.File.file_read("movies")
    movie_dict = dict()
    count = 0
    for item in movie_file:
        m = Movie(movie_file[item]["title"], movie_file[item]["foreign_title"], movie_file[item]["director"],
                  movie_file[item]["age_rating"],
                  movie_file[item]["duration"], movie_file[item]["is_3d"], movie_file[item]["image"])
        if m.movie_is_3d():
            movie_dict[item] = vars(m)
            print("Movie found! Title:", m.movie_name(), ", foreign title: ", m.movie_foreign_title(), ", director: ",
                  m.movie_director(), ", age rating: ", m.movie_age_rating(), ", duration: ", m.movie_duration(),
                  ", is in 3D: ", m.movie_is_3d(), ", image link:", m.movie_image())

            count = count + 1
    filereadwrite.File.file_write("results", movie_dict)
    if count == 0:
        print("No 3D movies found!")
    else:
        print("3D movies found: ", count)


# Only movies specified by the age rating are printed.
def output_movies_age_rating(rating, criterion):
    movie_file = filereadwrite.File.file_read("movies")
    movie_dict = dict()
    count = 0
    for item in movie_file:
        m = Movie(movie_file[item]["title"], movie_file[item]["foreign_title"], movie_file[item]["director"],
                  movie_file[item]["age_rating"],
                  movie_file[item]["duration"], movie_file[item]["is_3d"], movie_file[item]["image"])

        if criterion == "Up to" or criterion == "up to" or criterion == "up" or criterion == "Up":
            if movie_file[item]["age_rating"] == "Nenurodyta":
                pass
            elif int(movie_file[item]["age_rating"]) < int(rating):
                movie_dict[item] = vars(m)
                print("Movie added! Title:", m.movie_name(), ", foreign title: ", m.movie_foreign_title(),
                      ", director: ", m.movie_director(), ", age rating: ", m.movie_age_rating(), ", duration: ",
                      m.movie_duration(), ", is in 3D: ", m.movie_is_3d(), m.movie_image())

                count = count + 1
                filereadwrite.File.file_write("results", movie_dict)
        if criterion == "From" or criterion == "from":
            if movie_file[item]["age_rating"] == "Nenurodyta":
                pass
            elif int(movie_file[item]["age_rating"]) >= int(rating):
                movie_dict[item] = vars(m)
                print("Movie added! Title:", m.movie_name(), ", foreign title: ", m.movie_foreign_title(),
                      ", director: ", m.movie_director(), ", age rating: ", m.movie_age_rating(), ", duration: ",
                      m.movie_duration(), ", is in 3D: ", m.movie_is_3d(), ", image link: ", m.movie_image())

                count = count + 1
                filereadwrite.File.file_write("results", movie_dict)
    if count == 0:
        print("No movies with such age rating found!")
    else:
        print("Movies ", criterion, " ", rating, " found: ", count)


# Output only movies that have been featured on the news.
def output_featured_movies():
    movie_found = False
    movie_dict = dict()
    movie_file = filereadwrite.File.file_read("movies")
    news_file = filereadwrite.File.file_read("news")

    for item in news_file:
        n = news.News(news_file[item]["title"], news_file[item]["description"], news_file[item]["link"])
        for item2 in movie_file:
            m = Movie(movie_file[item2]["title"], movie_file[item2]["foreign_title"], movie_file[item2]["director"],
                      movie_file[item2]["age_rating"],
                      movie_file[item2]["duration"], movie_file[item2]["is_3d"], movie_file[item2]["image"])

            if m.movie_name() in n.news_name() or m.movie_name() in n.news_description():
                movie_found = True
                movie_dict[item] = vars(m)
                print("Movie added! Title:", m.movie_name(), ", foreign title: ", m.movie_foreign_title(),
                      ", director: ", m.movie_director(), ", age rating: ", m.movie_age_rating(), ", duration: ",
                      m.movie_duration(), ", is in 3D: ", m.movie_is_3d(), ", image link: ", m.movie_image())

                filereadwrite.File.file_write("results", movie_dict)

    if not movie_found:
        print("No movies with such name found in the news!")


# Every movie, along with its attributes, is printed.
def output_movies():
    movie_dict = dict()
    movie_file = filereadwrite.File.file_read("movies")
    for item in movie_file:
        m = Movie(movie_file[item]["title"], movie_file[item]["foreign_title"], movie_file[item]["director"],
                  movie_file[item]["age_rating"],
                  movie_file[item]["duration"], movie_file[item]["is_3d"], movie_file[item]["image"])
        movie_dict[item] = vars(m)
        filereadwrite.File.file_write("results", movie_dict)
        print("Movie  title:", m.movie_name(), ", foreign title: ", m.movie_foreign_title(), ", director: ",
              m.movie_director(), ", age rating: ", m.movie_age_rating(), ", duration: ", m.movie_duration(),
              ", is in 3D: ", m.movie_is_3d(), ", image link:", m.movie_image(), ",")
